.travis/rpmbuild_test

```diff
@@ -2,31 +2,41 @@
 
 # this script runs as root under docker
 
-OS_VERSION="$1"
+deps_file=".travis/deps/${OS_TYPE}_deps"
+
+if [[ -e "${deps_file}" ]]; then
+    source "${deps_file}"
+else
+    echo "${OS_TYPE} not supported yet, please create ${deps_file}!"
+    exit 1
+fi
 
 # build and install
-yum install -y libtool rpm-build make yum-utils squashfs-tools libarchive-devel
+install_autogen_deps
 ./autogen.sh
+
+# the final spec file is only created after configure, but we need it before to
+# parse it for build dependencies.
+sed 's/@PACKAGE_VERSION@/0/' < singularity.spec.in > singularity_tmp.spec
+
+install_configure_deps
 ./configure
 make dist
 rpmbuild -ta *.tar.gz
-rpm -iv ~/rpmbuild/RPMS/*/*.rpm
+
+install_rpms
 
 if [ "$OS_VERSION" = 6 ]; then
     echo "the python tests have not yet been ported to centos6 python, and will be skipped."
-else
-    # Install python 3 and pylint
-    yum install -y epel-release
-    yum install -y python34 python34-pip python34-setuptools docker
-    yum install -y pylint
 fi
 
+install_test_deps
+
 # run the test suite
-yum install -y libtool sudo which e2fsprogs
 sed -i 's,^prefix=.*,prefix=/usr,' test.sh
 sed -i 's,^sysconfdir=.*,sysconfdir=/etc,' test.sh
 sed -i 's,^localstatedir=.*,localstatedir=/var,' test.sh
-useradd -u 1000 testuser
+useradd -u 1000 --create-home testuser
```

Comment:
On these lines, you should use the values from:
```
rpm -E '%{_prefix}'
rpm -E '%{_sysconfdir}'
rpm -E '%{_localstatedir}'
```

That will pull the expanded macro value out in case it ever changes... for some reason.

Reply:
Good idea! I'm all for it :+1:

Reply:
I've ... seen it. Usually I've needed to rip out the normal `%configure` macro, and use my own options, and or override the normal `_libexecdir`, `_prefix`, etc... 

But, yeah... I've sadly seen that before. :/